save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/OrganizationView.kt

```diff
@@ -268,17 +274,18 @@ class OrganizationView : AbstractView<OrganizationProps, OrganizationViewState>(
         displayModal(state.isErrorOpen, state.errorLabel, state.errorMessage, smallTransparentModalStyle, errorCloseCallback) {
             buttonBuilder(state.closeButtonLabel ?: "Close", "secondary") { errorCloseCallback() }
         }
-
-        displayModal(state.isConfirmWindowOpen, state.confirmLabel, state.confirmMessage, smallTransparentModalStyle, { setState { isConfirmWindowOpen = false } }) {
-            buttonBuilder("Ok") {
-                when (state.confirmationType) {
-                    ConfirmationType.DELETE_CONFIRM -> deleteOrganizationBuilder()
-                    else -> throw IllegalStateException("Not implemented yet")
+        withNavigate {navigateContext ->
+            displayModal(state.isConfirmWindowOpen, state.confirmLabel, state.confirmMessage, smallTransparentModalStyle, { setState { isConfirmWindowOpen = false } }) {
+                buttonBuilder("Ok") {
+                    when (state.confirmationType) {
+                        ConfirmationType.DELETE_CONFIRM -> navigateContext.deleteOrganizationBuilder()
+                        else -> throw IllegalStateException("Not implemented yet")
```

Comment:
What is this exception for? If you want to throw not implemented - use TODO

Reply:
Done